lesson10/homework/internal/ports/httpgin/user_handlers_test.go

```diff
@@ -0,0 +1,277 @@
+package httpgin
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+	"homework10/internal/adapters/userrepo"
+	"homework10/internal/entities/user"
+	ads1ServiceMock "homework10/internal/ports/httpgin/adsport/mocks"
+	user1ServiceMock "homework10/internal/ports/httpgin/userport/mocks"
+	"homework10/pkg/logger"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+var ErrUnexpected = errors.New("unexpected server error")
+
+type userData struct {
+	Id       int64  `json:"id"`
+	Nickname string `json:"nickname"`
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
+
+type userResponse struct {
+	Data userData `json:"data"`
+}
+
+type UserApiTestSuite struct {
```

Comment:
обычно пишут заглавными буквами слова  типа API, ID
```suggestion
type UserAPITestSuite struct {
```